### ISSUE TITLE
issues getting started

### ISSUE THREAD

**Issue by dcsan** [nfp-projects/nfp_www#6]
I was looking for something lightweight with mithril... and came here.

but this seems to have a LOT more stuff than your tutorial - google closure??

having problems

```
npm start
```

then open to `localhost:3000`
gives a plain blank grey screen and some console errors

two other connections seem to be opened:

![image](https://cloud.githubusercontent.com/assets/514002/13107860/b075b430-d524-11e5-96ec-f47d297228d9.png)

i ran `livereload`

```
npm run livereload

> nfp_www@0.1.0 livereload /Users/dc/dev/rikai/chateditor/mitty
> live-reload --port 9091 dist/

live reload server listening on port 9091 reloading on files
```

which gets some errors, but tsill getting a white screen and error message.

is this repo even in a stable state?

**Comment by TheThing** [comment]
I'm glad you asked.

You are right about one thing and that is, this project is not lightweight at all. Which is why I've decided to completely decimate this code and start over. I'm currently working on building the infrastructure for the new code as we speak.

Concerning the blank grey screen, one of things I did with this project is I separated the frontend and backend. This is the frontend but it requires the backend to be running: https://github.com/nfp-projects/nfp_api

After working on this project for quite some time now I noticed that this kind of thing, separation between frontend and backend, was a really bad decision on my half. Makes the project tons more complicated to work with, run, deploy and etc.

But the repo is in stable state or should be. Just requires nfp_api to be running with a few data that I was stupid enough not to include in the project.

Also, what google closure tutorial?

**Comment by dcsan** [comment]
oh! is there any work in progress for the new slimmer version? very interested.

the point of using mithril (7kb!) is that the whole complex build chain should not be needed.
for dev, and even production maybe - why do you need the separate server as opposed to just serving everything from /public on the same express server?

A pre-deploy script to concat and gzip all the JS would be good enough?

> Also, what google closure tutorial?

you included closuercompiler in the deps.
https://github.com/nfp-projects/nfp_www/blob/master/package.json#L41
https://www.npmjs.com/package/closurecompiler

**Comment by TheThing** [comment]
> why do you need the separate server as opposed to just serving everything from /public on the same express server?

Exactly, it was a bad decision I did in the beginning and I regret it a lot now.

> A pre-deploy script to concat and gzip all the JS would be good enough?

I've thought about that but considering I don't want builded version in my github, I'm kinda forced to build the output on production. Plus with this handy guide: http://jonatan.nilsson.is/quick-start-guide-for-any-node-based-spa-with-mithril/ that is just a matter of running one npm command :)

And I include closurecompiler to "compile" my javascript to "better" javascript
